RecoLocalCalo/EcalRecProducers/plugins/EcalCPUUncalibRecHitProducer.cc

```diff
@@ -0,0 +1,200 @@
+        edm::ConfigurationDescriptions& confDesc) {
+    edm::ParameterSetDescription desc;
+
+    desc.add<edm::InputTag>("recHitsInLabelEB", 
+        edm::InputTag{"ecalUncalibRecHitProducerGPU", "EcalUncalibRecHitsEB"});
+    desc.add<edm::InputTag>("recHitsInLabelEE", 
+        edm::InputTag{"ecalUncalibRecHitProducerGPU", "EcalUncalibRecHitsEE"});
+    desc.add<std::string>("recHitsOutLabelEB", "EcalUncalibRecHitsEB");
+    desc.add<std::string>("recHitsOutLabelEE", "EcalUncalibRecHitsEE");
+    desc.add<bool>("containsTimingInformation", false);
+
+    std::string label = "ecalCPUUncalibRecHitProducer";
+    confDesc.add(label, desc);
+}
+
+EcalCPUUncalibRecHitProducer::EcalCPUUncalibRecHitProducer(
+        const edm::ParameterSet& ps) 
+    : recHitsInEBToken_{consumes<cms::cuda::Product<ecal::UncalibratedRecHit<ecal::Tag::ptr>>>(
+        ps.getParameter<edm::InputTag>("recHitsInLabelEB"))}
+    , recHitsInEEToken_{consumes<cms::cuda::Product<ecal::UncalibratedRecHit<ecal::Tag::ptr>>>(
+        ps.getParameter<edm::InputTag>("recHitsInLabelEE"))}
+    , recHitsOutEBToken_{produces<ecal::UncalibratedRecHit<ecal::Tag::soa>>(
+        ps.getParameter<std::string>("recHitsOutLabelEB"))}
+    , recHitsOutEEToken_{produces<ecal::UncalibratedRecHit<ecal::Tag::soa>>(
+        ps.getParameter<std::string>("recHitsOutLabelEE"))}
+    , containsTimingInformation_{ps.getParameter<bool>("containsTimingInformation")}
+{}
+
+EcalCPUUncalibRecHitProducer::~EcalCPUUncalibRecHitProducer() {}
+
+void EcalCPUUncalibRecHitProducer::acquire(
+        edm::Event const& event,
+        edm::EventSetup const& setup,
+        edm::WaitingTaskWithArenaHolder taskHolder) 
+{
+    // retrieve data/ctx
+    auto const& ebRecHitsProduct = event.get(recHitsInEBToken_);
+    auto const& eeRecHitsProduct = event.get(recHitsInEEToken_);
+    cms::cuda::ScopedContextAcquire ctx{ebRecHitsProduct, std::move(taskHolder)};
+    auto const& ebRecHits = ctx.get(ebRecHitsProduct);
+    auto const& eeRecHits = ctx.get(eeRecHitsProduct);
+
+    // resize the output buffers
+    recHitsEB_.resize(ebRecHits.size);
+    recHitsEE_.resize(eeRecHits.size);
+
+    // enqeue transfers
+    cudaCheck( cudaMemcpyAsync(recHitsEB_.did.data(),
+                               ebRecHits.did,
+                               recHitsEB_.did.size() * sizeof(uint32_t),
+                               cudaMemcpyDeviceToHost,
+                               ctx.stream()) );
+    cudaCheck( cudaMemcpyAsync(recHitsEE_.did.data(),
+                               eeRecHits.did,
+                               recHitsEE_.did.size() * sizeof(uint32_t),
+                               cudaMemcpyDeviceToHost,
+                               ctx.stream()) );
+
+    cudaCheck( cudaMemcpyAsync(recHitsEB_.amplitudesAll.data(),
+                               ebRecHits.amplitudesAll,
+                               recHitsEB_.amplitudesAll.size() * sizeof(uint32_t),
```

Comment:
shouldn't this be float?
Since the output of the local reconstruction is a float, not an int.

Similar comment for the lines below

Reply:
good point, i will push now